GroupGame/Assets/Scripts/CharacterMove.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(CharacterController))]
+
+public class CharacterMove : MonoBehaviour
+{
+    public float speed = 3.0F;          //the moving speed of the character
+
+    public float rotateSpeed = 3.0F;    //the rotate speed of the character
+
+    public float jumpSpeed = 8.0f;      //the jump force of the character
+
+    public float gravity = 20.0f;       //the force of gravity on the character
+
+    public float GroundOffset = .2f;    //the offset for the IsGrounded check. Useful for recognizing slopes and imperfect ground.
+
+    private Vector3 moveDirection = Vector3.zero;   //the direction the character should move.
+
+
+    ///The check to see if the character is currently on the ground.
+    private bool isGrounded(){
+        RaycastHit hit;
+        Physics.Raycast(this.transform.position, -this.transform.up,out hit, 10);   //A short ray shot directly downward from the center of the character.
+
+        if(hit.distance == 0)                                               //if the distance is zero, the ray probably did not hit anything.
```

Comment:
Since this is a float comparison, could we play it safe and do:

``
if(Math.abs(hit.distance) < Single.Epsilon)
``